lib/nrf_modem_lib/Kconfig.modemlib

```
@@ -132,6 +137,18 @@ config NRF_MODEM_LIB_SHMEM_TRACE_SIZE
 	help
 	  Size of the shared memory area used to receive modem traces.
 
+if NRF_MODEM_LIB_TRACE_ENABLED
+
+config NRF_MODEM_LIB_TRACE_HEAP_SIZE
+	int "Modem trace heap size"
+	default 2048
+	range 2048 4096
+	help
+	  Size of the heap buffer used for modem traces.
+	  This heap is allocated in the application's RAM.
+
+endif # NRF_MODEM_LIB_TRACE_ENABLED
```

Right now `nrf_modem_os.c` is not compiling without `MODEM_LIB_TRACE_ENABLED=y` because `NRF_MODEM_LIB_TRACE_HEAP_SIZE` will be undeclared in this case.

I suggest we do the same with `NRF_MODEM_LIB_TRACE_HEAP_SIZE` as is done with `NRF_MODEM_LIB_SHMEM_TRACE_SIZE`. That is, always keep `NRF_MODEM_LIB_TRACE_HEAP_SIZE` visible, but reduce the size to zero in the event that `MODEM_LIB_TRACE_ENABLED` is not defined. What do you think @lemrey ?
```suggestion
config NRF_MODEM_LIB_TRACE_HEAP_SIZE
	int "Modem trace heap size"
	default 2048 if NRF_MODEM_LIB_TRACE_ENABLED
	default 0
	range 0 4096
	help
	  Size of the heap buffer used for modem traces.
	  This heap is allocated in the application's RAM.
```

---

It wasn't a problem when the trace heap was in the trace module, but didn't think about that when I moved it to `nrf_modem_os.c`. Updated to your suggestion now.